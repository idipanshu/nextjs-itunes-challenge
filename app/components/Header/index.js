/**
 *
 * Header
 *
 */

import React from 'react';
import { Layout } from 'antd';
import styled from 'styled-components';
import { injectIntl } from 'react-intl';
import { fonts, colors } from '@themes';
import Text from '@components/Text';

const StyledHeader = styled(Layout.Header)`
  && {
    &.ant-layout-header {
      padding: 0 1rem;
      height: 7rem;
    }
    display: flex;
    justify-content: center;
    background-color: ${colors.primary};
  }
`;
const Title = styled(Text)`
  && {
    margin-bottom: 0;
    ${fonts.style.heading};
    font-family: verdana, sans-serif;
    display: flex;
    align-self: center;
    color: ${colors.text};
  }
`;
function Header(props) {
  return (
    <StyledHeader {...props} data-testid="header">
      <Title type="heading" id="heading_text" />
    </StyledHeader>
  );
}

export default injectIntl(Header);
