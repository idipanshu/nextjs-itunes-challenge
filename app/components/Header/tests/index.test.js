import React from 'react';
import { renderProvider } from '@utils/testUtils';
import Header from '../index';

describe('<Header />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(<Header />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should have a page header', () => {
    const { getByTestId } = renderProvider(<Header />);
    expect(getByTestId('header-title').textContent).toEqual('Search What You Like');
  });
});
