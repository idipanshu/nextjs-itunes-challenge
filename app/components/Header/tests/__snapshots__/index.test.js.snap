// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render and match the snapshot 1`] = `
.c1 {
  white-space: pre-line;
}

.c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #b59862;
}

.c0.c0.ant-layout-header {
  padding: 0 1rem;
  height: 7rem;
}

.c2.c2 {
  margin-bottom: 0;
  font-size: 2.25rem;
  font-weight: 500;
  font-family: verdana,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #020826;
}

<body>
  <div>
    <header
      class="ant-layout-header c0"
      data-testid="header"
      intl="[object Object]"
    >
      <span
        class="c1 c2"
        data-testid="header-title"
        type="heading"
      >
        Search What You Like
      </span>
    </header>
  </div>
</body>
`;
